extensions/interactions/MultipleChoiceInput/protractor.js

```diff
@@ -38,20 +38,19 @@ var expectInteractionDetailsToMatch = function(
     elem, richTextInstructionsArray) {
   elem.all(by.repeater('choice in $ctrl.choices'))
     .then(function(optionElements) {
-      var choices = [];
       expect(optionElements.length).toEqual(richTextInstructionsArray.length);
+      var promises = [];
       for (var i = 0; i < optionElements.length; i++) {
-        optionElements[i].element(by.css(
-          '.protractor-test-multiple-choice-option'
-        )).getText().then(function(text) {
-          choices.push(text);
-          if (choices.length === richTextInstructionsArray.length) {
-            for (var j = 0; j < choices.length; j++) {
-              expect(richTextInstructionsArray).toContain(choices[j]);
-            }
-          }
-        });
+        promises.push(optionElements[i].element(by.css(
+          '.protractor-test-multiple-choice-option')).getText());
       }
+      protractor.promise.all(promises).then(function(results) {
+        richTextInstructionsArray.sort();
```

Comment:
Never modify an array that is passed into a function. Make a copy if needed (using angular.copy or equivalent).

Otherwise this may mess up things for whoever is calling the function, since the array will remain sorted after this function returns.

Reply:
I am sorry for not realizing this silly mistake. Done.